x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/monitor_add_page.tsx

```diff
@@ -6,33 +6,44 @@
  */
 
 import React, { useEffect } from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { useTrackPageview } from '@kbn/observability-plugin/public';
 
-import { getServiceLocations } from '../../state';
+import { getServiceLocations, selectServiceLocationsState } from '../../state';
 import { ServiceAllowedWrapper } from '../common/wrappers/service_allowed_wrapper';
 
 import { useKibanaSpace } from './hooks';
 import { MonitorSteps } from './steps';
 import { MonitorForm } from './form';
+import { LocationsLoadingError } from './locations_loading_error';
 import { ADD_MONITOR_STEPS } from './steps/step_config';
 import { useMonitorAddEditBreadcrumbs } from './use_breadcrumbs';
+import { LoadingState } from '../monitors_page/overview/overview/monitor_detail_flyout';
 
-const MonitorAddPage = () => {
+export const MonitorAddPage = () => {
   useTrackPageview({ app: 'synthetics', path: 'add-monitor' });
   const { space } = useKibanaSpace();
   useTrackPageview({ app: 'synthetics', path: 'add-monitor', delay: 15000 });
   useMonitorAddEditBreadcrumbs();
   const dispatch = useDispatch();
+  const { locationsLoaded, error: locationsError } = useSelector(selectServiceLocationsState);
 
   useEffect(() => {
-    dispatch(getServiceLocations());
-  }, [dispatch]);
+    if (!locationsLoaded) {
```

Comment:
does it accounts for if locations are already loaded. And use adds anew location and comes back to this UI?

Reply:
It doesn't, actually. Thanks for asking. I was under the impression that adding a private location essentially triggered another load of locations, and refreshed the redux state, but it doesn't.

I looked into updating that logic, but it needs to be overhauled entirely because of https://github.com/elastic/kibana/issues/151762, so I opted out of doing that for now given that the logic will look very different soon.